pkg/cephconfig/config_test.go

```diff
@@ -0,0 +1,813 @@
+			},
+			searchText: "cache",
+			expected:   true,
+		},
+		{
+			name: "match in daemon default",
+			info: ConfigParamInfo{
+				Name:          "osd.0",
+				DaemonDefault: "cache_size=1G",
+			},
+			searchText: "cache",
+			expected:   true,
+		},
+		{
+			name: "case insensitive match",
+			info: ConfigParamInfo{
+				Name: "OSD.0",
+			},
+			searchText: "osd",
+			expected:   true,
+		},
+		{
+			name: "no match",
+			info: ConfigParamInfo{
+				Name: "osd.0",
+			},
+			searchText: "mon",
+			expected:   false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := matchesFullText(tt.info, strings.ToLower(tt.searchText))
+			req.Equal(tt.expected, result)
+		})
+	}
+}
+
+func TestMatchesType(t *testing.T) {
+	strType := pb.ConfigParam_str
+	intType := pb.ConfigParam_int
+	unknownType := pb.ConfigParam_ParamType(999)
+
+	tests := []struct {
+		name      string
+		info      ConfigParamInfo
+		paramType *pb.ConfigParam_ParamType
+		expected  bool
+	}{
+		{
+			name: "nil type should match any",
+			info: ConfigParamInfo{
+				Type: "str",
+			},
+			paramType: nil,
+			expected:  true,
+		},
+		{
+			name: "matching type",
+			info: ConfigParamInfo{
+				Type: "str",
+			},
+			paramType: &strType,
+			expected:  true,
+		},
+		{
+			name: "case insensitive match",
+			info: ConfigParamInfo{
+				Type: "INT",
+			},
+			paramType: &intType,
+			expected:  true,
+		},
+		{
+			name: "non-matching type",
+			info: ConfigParamInfo{
+				Type: "bool",
+			},
+			paramType: &strType,
+			expected:  false,
+		},
+		{
+			name: "unknown enum type",
+			info: ConfigParamInfo{
+				Type: "unknown",
+			},
+			paramType: &unknownType,
+			expected:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := matchesType(tt.info, tt.paramType)
+			if result != tt.expected {
+				t.Errorf("matchesType() = %v, want %v", result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestParseMinMax(t *testing.T) {
+	req := require.New(t)
+	tests := []struct {
+		name string
+		in   interface{}
+		want *float64
+	}{
+		{"string zero ('0')", "0", floatPtr(0)},
+		{"empty string", "", nil},
+		{"float64 positive (1.1)", 1.1, floatPtr(1.1)},
+		{"float64 negative (-1.1)", -1.1, floatPtr(-1.1)},
+		{"string negative integer ('-1')", "-1", floatPtr(-1)},
+		{"string positive integer ('1')", "1", floatPtr(1)},
+		{"string positive float ('1.1')", "1.1", floatPtr(1.1)},
+		{"string negative float ('-1.1')", "-1.1", floatPtr(-1.1)},
+		{"nil value", nil, nil},
+		{"unsupported type (struct{})", struct{}{}, nil},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ParseMinMax(tt.in)
+			if tt.want == nil {
+				req.Nil(got, "expected nil, got %v", got)
+			} else {
+				req.NotNil(got, "expected %v, got nil", *tt.want)
+				req.Equal(*tt.want, *got, "expected %v, got %v", *tt.want, *got)
+			}
+		})
+	}
+}
+
+func TestMergeParams(t *testing.T) {
```

Comment:
this function is a good candidate of table test.
you have 2 args(base, cluster) and result.
then you check that names in result are exactly equal to cluster.

it is hard to review current test but it looks like some use cases are missing: there can be new values at start, in the middle and in the end of cluster array. the same for missing values. and combination of new and missing values